arangod/RestHandler/RestAqlUserFunctionsHandler.cpp

```diff
@@ -110,32 +108,39 @@ RestStatus RestAqlUserFunctionsHandler::execute() {
       generateError(res);
     }
     return RestStatus::DONE;
-  } // DELETE
-  else if (type == rest::RequestType::GET) {
+    // DELETE
+  } else if (type == rest::RequestType::GET) {
     // JSF_get_api_aqlfunction.md
     // GET /_api/aqlfunction - figure out parameters - function namespace
     std::string functionNamespace;
     std::vector<std::string> const& suffixes = _request->decodedSuffixes();
 
-    
-    if ((suffixes.size() != 1) || suffixes[0].empty() ) {
+    if (!suffixes.empty()) {
+      if ((suffixes.size() != 1) || suffixes[0].empty() ) {
+        extractStringParameter(StaticStrings::Prefix, functionNamespace);
+        if (functionNamespace.empty()) {
+          generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_SUPERFLUOUS_SUFFICES,
+                        "superfluous suffix, expecting _api/aqlfunction/[<functionname or prefix>|?" +
+                        StaticStrings::Prefix + "=<functionname or prefix>]");
+          return RestStatus::DONE;
+        }
+      } else {
+        functionNamespace = suffixes[0];
+      }
+    } else {
       extractStringParameter(StaticStrings::Prefix, functionNamespace);
       if (functionNamespace.empty()) {
-        generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_SUPERFLUOUS_SUFFICES,
-                      "superfluous suffix, expecting _api/aqlfunction/[<functionname or prefix>|?" +
-                      StaticStrings::Prefix + "=<functionname or prefix>]");
-        return RestStatus::DONE;
+        // compatibility mode
+        extractStringParameter("namespace", functionNamespace);
```

Comment:
shouldn't we go via a StaticStrings::...?

Reply:
yes, we should.